flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamGroupWindowAggregate.scala

```diff
@@ -131,6 +135,19 @@ class DataStreamGroupWindowAggregate(
           "non-windowed GroupBy aggregation.")
     }
 
+    val isCountWindow = window match {
+      case TumblingGroupWindow(_, _, size) if isRowCountLiteral(size) => true
+      case SlidingGroupWindow(_, _, size, _) if isRowCountLiteral(size) => true
+      case _ => false
+    }
+
+    if (isCountWindow && grouping.length > 0 && queryConfig.getMinIdleStateRetentionTime < 0) {
+      LOG.warn(
```

Comment:
I think this can be argued in both ways. If we log an error message could also throw an exception and prevent the program to be executed. However, there are also other strategies to deal with state cleanup, e.g., by configuring RocksDB.